src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -289,6 +289,7 @@ unique_ptr<BoundQueryNode> Binder::BindNode(SelectNode &statement) {
 	// create a mapping of (alias -> index) and a mapping of (Expression -> index) for the SELECT list
 	case_insensitive_map_t<idx_t> alias_map;
 	expression_map_t<idx_t> projection_map;
+	this->column_alias_binder = make_unique<ColumnAliasBinder>(*result, alias_map);
```

Comment:
this is a little dodgy.. it's a member var that is only used by BindNode. The refactoring required to pass it through args was pretty high, so I did this instead.
It would probably be ok (?) if it didn't require the reset on #307, but that has issues as well because `alias_map` is local to `BindNode`, so ColumnAliasBinder can outlive its `alias_map`.

Options
 - current behaviour
 - make CAB a member of the BoundSelectNode?
 - make CAB own alias_map, or use shared_ptr, or something, so it can stay a member of this forever and not need manual cleanup
 - pass CAB through QualifyColumnNames and children as argument (ew)
 - ???

Reply:
Agreed - I don't think this should be in the binder at all. Can we perhaps override `QualifyColumnName` in the `SelectBinder`, and store the column alias binder in the `SelectBinder` only? Then instead of using the generic `ExpressionBinder::QualifyColumnNames` we use `select_binder.QualifyColumnNames` explicitly in this one location.

Reply:
Thanks for giving this some thought, I'll give that a shot.